tests/storage/restore.go

```diff
@@ -1493,6 +1493,133 @@ var _ = SIGDescribe("VirtualMachineRestore Tests", func() {
 					Entry("to a new VM", true),
 				)
 			})
+
+			Context("with cross namespace clone ability", func() {
+				var sourceDV *cdiv1.DataVolume
+				var cloneRole *rbacv1.Role
+				var cloneRoleBinding *rbacv1.RoleBinding
+
+				BeforeEach(func() {
+					sourceSC, exists := libstorage.GetRWOFileSystemStorageClass()
+					if !exists || sourceSC == snapshotStorageClass {
+						Skip("Two storageclasses required for this test")
+					}
+
+					source := libstorage.NewRandomDataVolumeWithRegistryImportInStorageClass(
+						cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros),
+						testsuite.NamespaceTestAlternative,
+						sourceSC,
+						corev1.ReadWriteOnce,
+						corev1.PersistentVolumeFilesystem,
+					)
+					if source.Annotations == nil {
+						source.Annotations = make(map[string]string)
+					}
+					source.Annotations["cdi.kubevirt.io/storage.bind.immediate.requested"] = "true"
+
+					source, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(source.Namespace).Create(context.Background(), source, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					sourceDV = source
+					sourceDV = waitDVReady(sourceDV)
+
+					role, roleBinding := libstorage.GoldenImageRBAC(testsuite.NamespaceTestAlternative)
+					role, err = virtClient.RbacV1().Roles(role.Namespace).Create(context.TODO(), role, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					cloneRole = role
+					roleBinding, err = virtClient.RbacV1().RoleBindings(roleBinding.Namespace).Create(context.TODO(), roleBinding, metav1.CreateOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					cloneRoleBinding = roleBinding
+				})
+
+				AfterEach(func() {
+					if sourceDV != nil {
+						err := virtClient.CdiClient().CdiV1beta1().DataVolumes(sourceDV.Namespace).Delete(context.TODO(), sourceDV.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+
+					if cloneRole != nil {
+						err := virtClient.RbacV1().Roles(cloneRole.Namespace).Delete(context.TODO(), cloneRole.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+					if cloneRoleBinding != nil {
+						err = virtClient.RbacV1().RoleBindings(cloneRoleBinding.Namespace).Delete(context.TODO(), cloneRoleBinding.Name, metav1.DeleteOptions{})
+						Expect(err).ToNot(HaveOccurred())
+					}
+				})
+
+				checkNoCloneAnnotations := func(vm *v1.VirtualMachine, shouldExist bool) {
+					pvcName := ""
+					for _, v := range vm.Spec.Template.Spec.Volumes {
+						if v.DataVolume != nil {
```

Comment:
nit: We could use here the function [PVCNameFromVirtVolume](https://github.com/kubevirt/kubevirt/blob/main/pkg/util/types/pvc.go#L72)